build-data-capturing-gradle-samples/capture-plugin-version/gradle-capture-plugin-version.gradle

```diff
@@ -0,0 +1,17 @@
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+
+/**
+ * This Gradle script captures the system properties available to each Test task, hashes the properties' values,
+ * and adds these as custom values.
+ * <code> apply from: file('capture-plugin-version.gradle') </code>
+ */
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+    return
+}
+
+URL url = com.gradle.scan.plugin.BuildScanPlugin.class.getClassLoader().getResource("com.gradle.scan.plugin.internal.meta.buildAgentVersion.txt")
+String buildAgentVersion = url.getText()
```

Comment:
Given this is Groovy DSL, the two lines above can use property-like access instead of getters.

Reply:
True. I do not feel too strongly, but I think people will be able to make the transition quite easily.